### ISSUE TITLE
Unable to load distro implementation with python 3.9

### ISSUE THREAD

**Issue by yuxisun1217** [Azure/WALinuxAgent#2079]
Description of problem:
WALA reports "WARNING Daemon Daemon Unable to load distro implementation for rhel. Using default distro implementation instead."
The root cause is the python version is 3.9.0, which doesn't have "platform.linux_distribution()" function. WALA calls distro.linux_distribution(full_distribution_name=False) instead. It returns "rhel" as distro_name, which doesn't match any item in azurelinuxagent/common/osutil/factory.py get_osutil() function.

Version-Release number of selected components (if applicable):
WALinuxAgent-2.2.48

How reproducible:
100%

Steps to Reproduce:
1. Create a VM on Azure with python 3.9 install and use WALA as provisioning agent.
2. Check /var/log/waagent.log

Actual results:
2020-11-16T13:30:02.897191Z INFO Daemon Daemon Azure Linux Agent Version:2.2.48.1
2020-11-16T13:30:02.911655Z INFO Daemon Daemon OS: rhel 9.0
2020-11-16T13:30:02.919980Z INFO Daemon Daemon Python: 3.9.0
2020-11-16T13:30:02.928820Z INFO Daemon Daemon Run daemon
2020-11-16T13:30:02.932309Z WARNING Daemon Daemon Unable to load distro implementation for rhel. Using default distro implementation instead.

Expected results:
Support both "redhat" and "rhel" distro names.

**Comment by pgombar** [comment]
Hi @yuxisun1217, you are right that Python 3.9 doesn't have `platform`, so the Agent uses `distro` instead.

Our `setup.py` script has `distro` listed as a dependency for py >= 3.7.
Can you confirm this dependency is broken if you install the agent using ` sudo python3.9 setup.py install --register-service`?

**Comment by yuxisun1217** [comment]
Hi @pgombar ,
I'm not quite sure if I understand it correctly. I ran `python3.9 setup.py install --register-service` and it worked well. 
The "distro" module exists. But the distro_name it returned is different from platform module.
```
>>> platform.linux_distribution(full_distribution_name=False)[0]
'redhat'
``` 
```
>>> distro.linux_distribution(full_distribution_name=False)[0]
'rhel'
```

**Comment by pgombar** [comment]
Ah sorry, my mistake. Due to the title I believed `distro` wasn't getting called at all. I see now that the issue you're raising is that `distro` returns "rhel" which we don't parse correctly as RedHat. Thanks for reporting this, we will fix it.

**Comment by pgombar** [comment]
Will be fixed in #2083
